nemo/collections/asr/models/ctc_models.py

```diff
@@ -44,8 +44,8 @@ class EncDecCTCModelConfig(ModelPTConfig):
 class EncDecCTCModel(ASRModel):
     """Encoder decoder CTC-based models."""
 
-    def __init__(self, cfg: EncDecCTCModelConfig):
-        super().__init__(cfg=cfg)
+    def __init__(self, cfg: EncDecCTCModelConfig, trainer=None):
```

Comment:
Changed the init of EncDecCTCModel. I don't know if we want this since we wanted all models to be initialized with a DictConfig object. But I cannot figure out how to pass a python class with Hydra

Reply:
this was in my original draft, but I've reverted it back based on review feedback :) 
Do you really need to pass trainer instance? Or this https://github.com/NVIDIA/NeMo/blob/candidate/examples/asr/speech_to_text.py#L69 is fine?

Reply:
PTL accepts all of the following for --gpu
```
# int: train on 2 gpus
trainer = Trainer(gpus=2)
# list: train on GPUs 1, 4 (by bus ordering)
trainer = Trainer(gpus=[1, 4])
trainer = Trainer(gpus='1, 4') # equivalent
# -1: train on all gpus
trainer = Trainer(gpus=-1)
trainer = Trainer(gpus='-1') # equivalent
# combine with num_nodes to train on multiple GPUs across nodes
# uses 8 gpus in total
trainer = Trainer(gpus=2, num_nodes=4)
``
I assume we want to support all of that. Now do we want to copy their initialize function and parse all of that, or do we just want to call trainer.init() and let them parse it and use trainer.num_gpus?

Reply:
I suppose its fine to pass in Trainer like this as it defaults to None anyway. we definitely dont want to cut-paste their logic, in case they change in the future (quite likely)

Reply:
why do we need this "parsing"? cannot we can simply call:
```
python my_script.py trainer. gpus=2
```
?

Reply:
@tkornuta-nvidia, how do I specify that I want to run on the 2nd gpu only? Or on the 3rd and 4th gpu only? How do we support `Trainer(gpus=[1, 4])`?

Reply:
If you are asking about Union, I am investigating it - but seems that OmegaConf doesn't support them...